builds.js

```diff
@@ -9,112 +12,327 @@ const environments = new Map([
   ['dist', 'prod'],
   ['prod', 'prod']
 ]);
+const defaultCacheRefreshInterval = 20 * 60 * 60 * 1000;
+const defaultCacheRefreshLimit = 10;
 
 /**
  * Build interface for the Warehouse API.
  *
- * @constructor
- * @param {Warehouse} warehouse Reference to the Warehouse instance.
- * @api public
+ * @class Builds
  */
-function Builds(warehouse) {
-  this.warehouse = warehouse;
-}
+class Builds {
+  /**
+   * @constructor
+   * @param {Warehouse} warehouse Reference to the Warehouse instance.
+   * @param {Object} [options] Warehouse options object
+   * @param {Object} [options.buildCache] An options object for caching build information
+   * @param {boolean} [options.buildCache.enabled] True if you want to use a cache of build information,
+   * False if you want to hit the service every time
+   * @param {number} [options.buildCache.refreshLimit] The number of cache entries that can be refreshed simultaneously.
+   * @param {number} [options.buildCache.refreshInterval] An interval in ms on which the cache should be refreshed.
+   * If the refresh takes longer than the interval, intervals will be skipped. A non-positive number results in the
+   * cache _never_ being refreshed
+   * @public
+   */
+  constructor(warehouse, options) {
+    this.warehouse = warehouse;
+    this._cacheRefreshIntervalId = null;
 
-/**
- * Get build information.
- *
- * @param {Object} params Parameters that specify pkg, env, version and/or locale.
- * @param {Function} fn Completion callback.
- * @returns {Warehouse} fluent interface.
- * @api public
- */
-Builds.prototype.get = function get(params, fn) {
-  const env = environments.get(params.environment || params.env) || 'dev';
-  const version = params.v || params.version;
-  const meta = params.meta ? 'meta' : null;
-  const locale = params.locale || 'en-US';
-  let pkg = params.package || params.pkg;
-
-  if (!pkg || !env) {
-    return fn(
-      new Error('invalid parameters supplied, missing `pkg` or `env`')
-    );
+    options = options || {};
+    if (options.buildCache && options.buildCache.enabled) {
+      this._refreshCache = this._refreshCache.bind(this);
+
+      /**
+       * @member {Map} cache The build cache
+       * @private
+       */
+      this._cache = new Map();
+
+      /**
+       * @member {number} _cacheRefreshLimit The number of cache entries that can be simultaneously refreshed
+       * @private
+       */
+      this._cacheRefreshLimit = options.buildCache.refreshLimit || defaultCacheRefreshLimit;
+
+      /**
+       * @member {number} _cacheRefreshInterval The interval in ms on which the cache is refreshed refreshed
+       * @private
+       */
+      this._cacheRefreshInterval = (options.buildCache.refreshInterval || options.buildCache.refreshInterval === 0) ?
```

Comment:
i dont think we need to handle the `0` case since that should never be set as the `interval` as its super impractical. So the above options setting change should be sufficient.

Reply:
Simply truthyiness check will get you want you want here as well:

``` js
this._cacheRefreshInterval = options.buildCache.refreshInterval || defaultCacheRefreshInterval;
```

Reply:
This is what I started with and I'm fine going back to it if we're okay with refreshinterval: 0 being sent in results in default and not _no_ refresh, which is what I was aiming for (since we still have to technically deal with negatives, should 0 result in no refresh or default)